src/defineMacro.js

```diff
@@ -116,3 +128,24 @@ export const _macros: MacroMap = {};
 export default function defineMacro(name: string, body: MacroDefinition) {
     _macros[name] = body;
 }
+
+export const _conditionals: {[string]: true} = {};
+
+export function defineConditional(
+    name: string,
+    evaluate: MacroContextInterface => boolean
+) {
+    _conditionals[name] = true;
+    // When an \if... is expanded, TeX reads ahead as far as necessary
+    // to determine whether the condition is true or false; and if false,
+    // it skips ahead until finding the \else, \or, or \fi that ends the
+    // skipped text.
+    defineMacro(name, function(context) {
+        const condition = evaluate(context);
+        context.conditions.push(condition);
+        if (!condition) {
+            context.skipConditionalText();
+        }
+        return '';
```

Comment:
Would `return []` be more efficient here, because it doesn't engage the tokenizer?  Not sure.  (Also applies to a lot of other macros.)

Alternatively, I wonder if it makes sense to check for `undefined` as a return value from a macro, and expand it to nothing in that case.  Then this line could be omitted, here and elsewhere.